extensions/lifecycle/tasks/LifecycleTask.js

```diff
@@ -586,6 +581,126 @@ class LifecycleTask extends BackbeatTask {
         return Math.floor(diff / (1000 * 60 * 60 * 24));
     }
 
+    /**
+     * Helper method to determine if a version is a Delete Marker
+     * @param {Object} version - single version object
+     * @return {boolean} true/false
+     */
+    _isDeleteMarker(version) {
+        // if no ETag, Size, and StorageClass, then it is a Delete Marker
+        return (
+            !Object.prototype.hasOwnProperty.call(version, 'ETag') &&
+            !Object.prototype.hasOwnProperty.call(version, 'Size') &&
+            !Object.prototype.hasOwnProperty.call(version, 'StorageClass')
+        );
+    }
+
+    /**
+     * Helper method for Expiration.Date and Expiration.Days rules
+     * Check if Expiration rules apply on the object or version
+     * @param {object} bucketData - bucket data
+     * @param {object} obj - single object or version
+     * @param {string} obj.LastModified - last modified date of object
+     * @param {object} rules - most applicable rules from `_getApplicableRules`
+     * @param {Logger.newRequestLogger} log - logger object
+     * @return {boolean} used to know if a rule has been applied already.
+     *   If this rule has been applied to the object or version, other rules
+     *   (i.e. transition) should not apply.
+     */
+    _checkAndApplyExpirationRule(bucketData, obj, rules, log) {
+        const daysSinceInitiated = this._findDaysSince(
+            new Date(obj.LastModified));
+
+        if (rules.Expiration.Date &&
+        rules.Expiration.Date < Date.now()) {
+            // expiration date passed for this object
+            const entry = {
+                action: 'deleteObject',
+                target: {
+                    owner: bucketData.target.owner,
+                    bucket: bucketData.target.bucket,
+                    key: obj.Key,
+                },
+                details: {
+                    lastModified: obj.LastModified,
+                },
+            };
+            this.sendObjectEntry(entry, err => {
+                if (!err) {
+                    log.debug('sent object entry for consumption', {
+                        method: 'LifecycleTask._compareObject',
+                        entry,
+                    });
+                }
+            });
+            return true;
+        }
+        if (rules.Expiration.Days !== undefined &&
```

Comment:
can just say `if (rules.Expiration.Days && ...` like in the condition above, right?

Reply:
Ah I realized I check for undefined because test cases can have Days set to 0. Aws only allows a minimum of 1 day to be set on lifecycle rules, but to test out if the rules are applied correctly, I have to set the days down to 0 to make sure that the kafka entries are added.